core/src/main/java/org/sonarsource/sonarlint/core/telemetry/TelemetryClient.java

```diff
@@ -53,15 +59,19 @@ void upload(TelemetryData data) {
     try {
       sendPost(httpFactory.buildClient(clientConfig), createPayload(data));
     } catch (Exception e) {
-      // fail silently
+      if (SonarLintUtils.isInternalDebugEnabled()) {
+        LOG.debug("Failed to upload telemetry data", e);
```

Comment:
Maybe it should be logged as an error? If the flag is enabled, means that whoever is usign it wants to see the errors. Debug might not appear at all if the SonarLint console doesn't have the debug flag activated.

Reply:
One reason we silently ignored these errors was obviously to avoid any kind of noise from telemetry for users. Imagine that reason doesn't exist. Would we log such errors at any level other than debug? I'm not sure.

But anyway, I will change all these to `error` level now, to make sure we see them if they ever happen. These should probably never happen, and when they do, we want to know.